src/coreclr/jit/lowerxarch.cpp

```diff
@@ -130,6 +130,21 @@ void Lowering::LowerStoreIndir(GenTreeIndir* node)
             return;
         }
     }
+    else if (!node->HasIndex() && node->gtGetOp2()->OperIs(GT_CNS_DBL) &&
+             (FloatingPointUtils::convertDoubleToUInt64(node->gtGetOp2()->AsDblCon()->gtDconVal) == 0))
+    {
+        // Replace *address = 0.0 with *address = 0 (integer) which is slightly more efficient.
+        const var_types intType = node->TypeIs(TYP_FLOAT) ? TYP_INT : TYP_LONG;
+        if (genTypeSize(intType) <= TARGET_POINTER_SIZE)
+        {
+            GenTreeIntCon* intCns = comp->gtNewIconNode(0, intType);
```

Comment:
TP nit: I would mutate the node in-place, via something like `ChangeOperConst`.

Reply:
Thanks, didn't know about that method 🙂